nlpretext/basic/preprocess.py

```diff
@@ -65,32 +82,95 @@ def lower_text(text: str):
     """
     return text.lower()
 
-def remove_stopwords(text: str, lang: str, custom_stopwords: list = None) -> str:
+
+def filter_groups(token: str, ignored_stopwords: list = None) -> str:
+    """
+    Given ``token`` str and a list of groups of words
+    that were concatenated into tokens, reverses the tokens
+    to their ungrouped state.
+
+    Parameters
+    ----------
+    token : string
+    ignored_stopwords : list of strings
+
+    Returns
+    -------
+    string
+    """
+    if ignored_stopwords:
+        for group in ignored_stopwords:
+            if token == remove_whitespace(group):
+                token = group
+    return token
+
+
+def ungroup_ignored_stopwords(tokens: list, ignored_stopwords: list = None) -> list:
+    """
+    Given ``tokens`` list of str and a list of groups of words
+    that are concatenated in tokens, reverses the tokens to
+    their ungrouped state.
+
+    Parameters
+    ----------
+    tokens : list of strings
+    ignored_stopwords : list of strings
+
+    Returns
+    -------
+    list of strings
+    """
+
+    return [filter_groups(token, ignored_stopwords) for token in tokens]
+
+
+def remove_stopwords(text: str, lang: str, custom_stopwords: list = None, ignored_stopwords: list = None) -> str:
     """
     Given ``text`` str, remove classic stopwords for a given language and
-    custom stopwords given as a list.
+    custom stopwords given as a list. Words and groups of words from
+    ignored_stopwords list are ignored during stopwords removal.
 
     Parameters
     ----------
     text : string
     lang : string
     custom_stopwords : list of strings
+    ignored_stopwords : list of strings
 
     Returns
     -------
     string
+
+    Raises
+    -------
+    ValueError
+        if ``custom_stopwords``  and ``ignored_stopwords`` have common elements.
     """
+    if custom_stopwords and ignored_stopwords:
+        if len(set(custom_stopwords) & set(ignored_stopwords)) > 0:
```

Comment:
Ce serait bien d'être plus informatif là dessus. On pourrait faire un 
`common_elements = set(custom_stopwords).intersection(set(ignored_stopwords))
if common_elements != set():
    raise ValueError(f"Found common words in custom_stopwords and ignored_stopwords: {common_elements}. Please remove duplicated values." )`